cmd/cephcsi.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2019 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/ceph/ceph-csi/pkg/cephfs"
+	"github.com/ceph/ceph-csi/pkg/rbd"
+	"github.com/ceph/ceph-csi/pkg/util"
+	"k8s.io/klog"
+)
+
+const (
+	rbdType    = "rbd"
+	cephfsType = "cephfs"
+
+	rbdDefaultName    = "rbd.csi.ceph.com"
+	cephfsDefaultName = "cephfs.csi.ceph.com"
+)
+
+var (
+	// common flags
+	vtype           = flag.String("type", "", "driver type [rbd|cephfs]")
```

Comment:
I think we need to update the deployment templates for this one, and also the files in deploy folder  for new flags

is this a breaking change? because once PR is merged new image will be pushed.

Reply:
Eventually yes, but I didn't plan to immediately deprecate the "split" images. Only new templates would be needed after the image was ready. I wasn't going to do that in this PR.